core/src/messaging/net_message.rs

```diff
@@ -132,6 +149,16 @@ impl NetMessage {
         &self.sender
     }
 
+    /// Returns the session of the `NetMessage`
+    pub fn session(&self) -> Option<SessionId> {
+        self.session
+    }
+
+    /// Sets the SessionId of the `NetMessage`
+    pub(crate) fn set_session(&mut self, session: SessionId) -> () {
+        self.session = Some(session);
+    }
```

Comment:
I actually don't necessary agree with Harald here. Remember that people could be writing 3rd-party network implementations? This makes it impossible for those to support session ids.